Products/DataCollector/zing/fact.py

```diff
@@ -69,55 +88,86 @@ def default(self, o):
             return o.args
         return JSONEncoder.default(self, o)
 
+
 FactEncoder = _FactEncoder()
 
 
-def serialize_datamap(device, dm, context):
-    """
-    Converts a datamap to a JSON-encoded list of facts.
-    """
+class FactContext(object):
+
+    def __init__(self, obj):
+        self.uuid = None
+        self.meta_type = None
+        self.name = None
+        self.mem_capacity = None
```

Comment:
Similar to earlier comment, it seems odd that we are capturing mem capacity as a context